source/agora/consensus/ValidatorSet.d

```diff
@@ -475,6 +475,10 @@ public class ValidatorSet
             return false;
         }
 
+        // Ignore same height pre-image because validators will gossip them
+        if (prev_preimage.distance == preimage.distance)
```

Comment:
Isn't it better to compare `if (prev_preimage == preimage)`? Because if the distance is the same but the hash is different we won't get any errors - but we should get the errors in the logs..

Reply:
Should we ? Does it really matter ?

Reply:
This addition is supposed to act as an additional filter for [`cb941ec`](https://github.com/bpfkorea/agora/commit/cb941ecab41a2c0256427289e054f286d1596c3e). The error for the wrong hash should be triggered by another validity check in `isInvalidReason()`. I will examine the code again.